### PR TITLE
Add log splitter binary for kubernetes alerts

### PR DESCRIPTION
**Pull request by vkgnosis**
After some discussion with devops we want to make this part of the
docker image instead of inserting it through configmap later.
Reimplemented in rust because we're a rust code base.

Once this is merged the plan is to change the kubernetes command to `sh -c "orderbook | log-splitter"`.

### Test Plan
new unit test

### CODE REVIEW

**Review by fleupold — APPROVED** [review]
Looks good, although I can how this is easily useful for any rust project that we run inside gnosis (e.g. even the IDO services, Safe gateway, etc).

Can we maybe publish this as a GPv2 independent binary crate? Otherwise I suspect people will just-copy paste it 😞

**Comment by vkgnosis** [comment]
That is a good point. Will make a new repo for it.

**Comment by vkgnosis** [comment]
Requesting review again because I changed this to pull from the other repository.

File: docker/Dockerfile.binary [hunk 1]
```diff
@@ -5,11 +5,18 @@ WORKDIR /usr/src/oba-services
 COPY . .
 RUN cargo build --target x86_64-unknown-linux-musl --release
 
+RUN \
+    cd .. && \
+    git clone https://github.com/gnosis/regex-stream-split.git && \
```
Comment (fleupold) [inline docker/Dockerfile.binary:10]: Should we consider pinning to a commit?

**Review by fleupold — APPROVED** [review]
Nice!

File: docker/Dockerfile.binary (hunk 1 above)
Comment (vkgnosis) [reply docker/Dockerfile.binary:10]: added

**Pull request merged**